doc/mapflags.txt

```diff
@@ -396,3 +396,10 @@ in '/conf/channels.conf'.
 Disables MVP tombs from appearing on a map.
 
 ---------------------------------------
+
+*hidehpbar
+
+Hides monster's HP bar on a map.
+Ignores config value of 'monster_hp_bars_info'.
```

Comment:
Judging from this description all of the ifs you added should check for the mapflag before the configuration.